libcaf_core/caf/serialization.test.cpp

```diff
@@ -0,0 +1,767 @@
+      if (parsed_value.size() > 5) {
+        CAF_RAISE_ERROR(std::logic_error, "invalid array size");
+      }
+      std::transform(parsed_value.cbegin(), parsed_value.cend(), iarray.begin(),
+                     [](const std::string& s) { return std::stoi(s); });
+      std::fill(iarray.begin() + parsed_value.size(), iarray.end(), 0);
+      return iarray;
+    }
+    if (type == "tuple") {
+      auto parsed_value = parse_value(value, ",");
+      if (parsed_value.size() != 3) {
+        CAF_RAISE_ERROR(std::logic_error, "invalid tuple size");
+      }
+      return std::tuple{std::stoi(parsed_value[0]), parsed_value[1],
+                        std::stoi(parsed_value[2])};
+    }
+    CAF_RAISE_ERROR(std::logic_error, "invalid type");
+  }
+
+  std::variant<int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t,
+               uint64_t, double, std::string, std::vector<int32_t>,
+               std::list<int32_t>, std::map<std::string, int32_t>,
+               std::unordered_map<std::string, int32_t>, std::set<int32_t>,
+               std::unordered_set<int32_t>, std::array<int32_t, 5>,
+               std::tuple<int32_t, std::string, int32_t>>
+  default_val(const std::string& type) {
+    if (type == "i8") {
+      return static_cast<int8_t>(0);
+    }
+    if (type == "i16") {
+      return static_cast<int16_t>(0);
+    }
+    if (type == "i32") {
+      return static_cast<int32_t>(0);
+    }
+    if (type == "i64") {
+      return static_cast<int64_t>(0);
+    }
+    if (type == "u8") {
+      return static_cast<uint8_t>(0);
+    }
+    if (type == "u16") {
+      return static_cast<uint16_t>(0);
+    }
+    if (type == "u32") {
+      return static_cast<uint32_t>(0);
+    }
+    if (type == "u64") {
+      return static_cast<uint64_t>(0);
+    }
+    if (type == "real") {
+      return 0.0;
+    }
+    if (type == "string") {
+      return "";
+    }
+    if (type == "vector") {
+      return std::vector<int32_t>{};
+    }
+    if (type == "list") {
+      return std::list<int32_t>{};
+    }
+    if (type == "map") {
+      return std::map<std::string, int32_t>{};
+    }
+    if (type == "umap") {
+      return std::unordered_map<std::string, int32_t>{};
+    }
+    if (type == "set") {
+      return std::set<int32_t>{};
+    }
+    if (type == "uset") {
+      return std::unordered_set<int32_t>{};
+    }
+    if (type == "array") {
+      return std::array<int32_t, 5>{};
+    }
+    if (type == "tuple") {
+      return std::tuple<int32_t, std::string, int32_t>{};
+    }
+    CAF_RAISE_ERROR(std::logic_error, "invalid type");
+  }
+
+private:
+  std::vector<std::string> parse_value(const std::string& list_val,
+                                       const std::string& delimiter) {
+    const std::regex delimiter_regex{delimiter};
+    return std::vector<std::string>{
+      std::sregex_token_iterator(list_val.cbegin(), list_val.cend(),
+                                 delimiter_regex, -1),
+      {}};
```

Comment:
I think using a `regex` here is overkill. Isn't this simply a `split`? https://github.com/actor-framework/actor-framework/blob/master/libcaf_core/caf/string_algorithms.hpp#L35

Reply:
Thanks, that is helpful. I was not aware of the existing split functions. I am working on updating the PR accordingly.